features/support/commit_helpers.rb

```diff
@@ -1,13 +1,141 @@
-# Returns the commits in the current directory
-def commits_in_repo keys = [:author, :file_content, :file_name, :message]
-  out = {}
+# Makes finding Git commits easier
+class CommitsFinder
```

Comment:
The procedural logic for this became so complex that I had to extract this into a helper class. The good news is that parts of this are so generic that it can be used to clean up many other areas and step definitions as well, hopefully leaving the code base in a better state than before.